esphome/components/ethernet/ethernet_component.cpp

```diff
@@ -52,30 +54,30 @@ void EthernetComponent::setup() {
 
   esp_eth_mac_t *mac = esp_eth_mac_new_esp32(&mac_config);
 
-  esp_eth_phy_t *phy;
+  phy_ = nullptr;
   switch (this->type_) {
     case ETHERNET_TYPE_LAN8720: {
-      phy = esp_eth_phy_new_lan87xx(&phy_config);
+      phy_ = esp_eth_phy_new_lan87xx(&phy_config);
```

Comment:
Same for all of these below too
```suggestion
      this->phy_ = esp_eth_phy_new_lan87xx(&phy_config);
```